src/main/java/com/cloudbees/jenkins/support/configfiles/SecretHandler.java

```diff
@@ -77,15 +81,37 @@ public void characters(char[] ch, int start, int length) throws SAXException {
                 super.characters(ch, start, length);
             }
         };
-        Source src = new SAXSource(xr, new InputSource(new StringReader(FileUtils.readFileToString(xmlFile))));
+        String str = FileUtils.readFileToString(xmlFile);
+        Source src = new SAXSource(xr, new InputSource(new StringReader(str)));
         final ByteArrayOutputStream result = new ByteArrayOutputStream();
         Result res = new StreamResult(result);
         Transformer transformer = TransformerFactory.newInstance().newTransformer();
         //omit xml declaration because of https://bugs.openjdk.java.net/browse/JDK-8035437
         transformer.setOutputProperty(OutputKeys.OMIT_XML_DECLARATION, "yes");
         transformer.setOutputProperty(OutputKeys.ENCODING, OUTPUT_ENCODING);
-        transformer.transform(src, res);
 
-        return result.toString("UTF-8");
+        try {
+            transformer.transform(src, res);
+            return result.toString("UTF-8");
+        } catch (TransformerException e) {
+            return findSecretFallback(str);
+        }
     }
+
+
+    private static String findSecretFallback(String xml) {
```

Comment:
This method definitely works as a fallback, but I am bit concerned about corner-cases. What if the file is damaged, e.g. what if the last `}` is missing? Does it add risks to this implementation?

Reply:
it is the same behavior than the current implementation. If it doesn't match with {}, the secret is not overwritten

Reply:
The current solution needs a valid XML, and it's hard to imagine a case when we get a corrupted secret and a full XML file in the case of the glitch. Anyway, it's rather nitpicking